Documentation/network/kubernetes/configuration.rst

```diff
@@ -150,65 +144,59 @@ The CNI auto installation is performed as follows:
+3. The file ``/etc/cni/net.d/05-cilium.conflist`` is written.
 
 
 Adjusting CNI configuration
 ---------------------------
 
-The CNI configuration file is automatically written and maintained by the
-scripts ``cni-install.sh`` and ``cni-uninstall.sh`` which are running as
-``postStart`` and ``preStop`` hooks of the Cilium pod.
-
-If you want to provide your own custom CNI configuration file, set the
-``CILIUM_CUSTOM_CNI_CONF`` environment variable to avoid overwriting the
-configuration file by adding the following to the ``env:`` section of the
-``cilium`` DaemonSet:
-
-.. code-block:: yaml
-
-        - name: CILIUM_CUSTOM_CNI_CONF
-          value: "true"
-
-The CNI installation can be configured with environment variables. These
-environment variables can be specified in the :term:`DaemonSet` file like this:
-
-.. code-block:: yaml
+cilium pod. It is written after the agent has finished initialization and
+is ready to handle pod sandbox creation. In addition, the agent will remove
+any other CNI configuration files by default.
+
+There are a number of Helm variables that adjust CNI configuration management.
+For a full description, see the helm documentation. A brief summary:
+
++--------------------+----------------------------------------+---------+
+| Helm variable      | Description                            | Default |
++====================+========================================+=========+
+| ``cni.customConf`` | Disable CNI configuration management   | false   |
++--------------------+----------------------------------------+---------+
+| ``cni.exclusive``  | Remove other CNI configuration files   | true    |
++--------------------+----------------------------------------+---------+
+| ``cni.install``    | Install CNI configuration and binaries | true    |
++--------------------+----------------------------------------+---------+
+
+
+If you want to provide your own custom CNI configuration file, you can do
+so by passing a path to a cni template file, either on disk or provided
+via a configMap. The Helm options that configure this are:
+
++----------------------+----------------------------------------------------------------+
+| Helm variable        | Description                                                    |
++======================+================================================================+
+| ``cni.readCniConf``  | Path (inside the agent) to a source CNI configuration file     |
++----------------------+----------------------------------------------------------------+
+| ``cni.configMap``    | Name of a ConfigMap containing a source CNI configuration file |
++----------------------+----------------------------------------------------------------+
+| ``cni.configMapKey`` | Install CNI configuration and binaries                         |
++----------------------+----------------------------------------------------------------+
+
+These Helm variables are converted to a smaller set of cilium ConfigMap keys:
```

Comment:
`cni-exclusive` is a settable helm value. Do you mean to make its value programmatically set?

Reply:
yeah.

Honestly, this whole document is just really old. We don't document any other helm variables in the same manner, but I was trying to preserve the existing style for CNI configuration.